src/DebugBar/Resources/widgets/mails/widget.js

```diff
@@ -14,30 +14,43 @@
 
         render: function() {
             this.$list = new  PhpDebugBar.Widgets.ListWidget({ itemRenderer: function(li, mail) {
-                $('<span />').addClass(csscls('subject')).text(mail.subject).appendTo(li);
-                $('<span />').addClass(csscls('to')).text(mail.to).appendTo(li);
-                if (mail.body) {
-                    li.click(function() {
-                        var popup = window.open('about:blank', 'Mail Preview', 'width=650,height=440,scrollbars=yes');
-                        var documentToWriteTo = popup.document;
-                        var headers = !mail.headers ? '' : $('<pre style="border: 1px solid #ddd; padding: 5px;" />')
-                            .append($('<code />').text(mail.headers)).prop('outerHTML');
-                        documentToWriteTo.open();
-                        documentToWriteTo.write(headers + mail.body);
-                        documentToWriteTo.close();
-                    });
-                } else if (mail.headers) {
-                    var headers = $('<pre />').addClass(csscls('headers')).appendTo(li);
-                    $('<code />').text(mail.headers).appendTo(headers);
-                    li.click(function() {
-                        if (headers.is(':visible')) {
-                            headers.hide();
-                        } else {
-                            headers.show();
-                        }
-                    });
-                }
-            }});
+                    $('<span />').addClass(csscls('subject')).text(mail.subject).appendTo(li);
+                    $('<span />').addClass(csscls('to')).text(mail.to).appendTo(li);
+                    if (mail.html || mail.html) {
+                        var header = $('<span />').addClass(csscls('filename')).text('');
+                        $('<a title="Mail Preview">View Mail</a>').on('click', function () {
+                            var popup = window.open('about:blank', 'Mail Preview', 'width=650,height=440,scrollbars=yes');
+                            var documentToWriteTo = popup.document;
+                            var headers = !mail.headers ? '' : $('<pre style="border: 1px solid #ddd; padding: 5px;" />')
+                                .append($('<code />').text(mail.headers));
+
+                            var body = $('<pre style="border: 1px solid #ddd; padding: 5px;" />').text(mail.body)
+                            var html = null;
+                            if (mail.html) {
```

Comment:
after `if (mail.html || mail.html) {`, this is always true

Reply:
Yes see other comment, should be when body is set but not html